apps/sms/desktop-mock/navigator_moz_mobilemessage.js

```diff
@@ -3,6 +3,18 @@
 
 'use strict';
 
+// Copied from Utils.Promise 
+function defer() {
```

Comment:
Copy the promise defer from Utils. We can still use Utils.Promise.defer but it's the only thing we need in this mock. I can reuse Utils.Promise.defer if you prefer.